### PR TITLE
Feature support pvc mount and fix filter bug

### PR DESCRIPTION
**Pull request by tiancandevloper**
Ⅰ. Describe what this PR does
1. support list pvc and which pod mount it
2. the old extend func use switch, it can be to long, and difficult to update it, so use factory to use it
3. old filter can only work with array[0], fix this bug

### CODE REVIEW

File: pkg/apiserver/cubeapi/resourcemanage/handle/extend.go [hunk 1]
```diff
@@ -48,6 +41,14 @@ import (
 	"github.com/kubecube-io/kubecube/pkg/utils/response"
 )
 
+var (
+	extendFuncMap map[string]ExtendFunc
```
Comment (weilaaa) [inline pkg/apiserver/cubeapi/resourcemanage/handle/extend.go:45]: just init map here: extendFuncMap = make(map[string]ExtendFunc)

File: pkg/apiserver/cubeapi/resourcemanage/resources/service/externalaccess.go [hunk 2]
```diff
@@ -17,20 +17,24 @@ limitations under the License.
 package service
 
 import (
+	"context"
+	errors2 "errors"
```
Comment (weilaaa) [inline pkg/apiserver/cubeapi/resourcemanage/resources/service/externalaccess.go:21]: do not rename standard lib

File: pkg/apiserver/cubeapi/resourcemanage/resources/service/externalaccess.go [hunk 3]
```diff
@@ -17,20 +17,24 @@ limitations under the License.
...
 	"fmt"
+	"net/http"
 	"strconv"
 	"strings"
 
-	"context"
-
 	jsoniter "github.com/json-iterator/go"
 	v1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/errors"
```
Comment (weilaaa) [inline pkg/apiserver/cubeapi/resourcemanage/resources/service/externalaccess.go:29]: rename it as apierrors

File: pkg/apiserver/cubeapi/resourcemanage/resources/cronjob/cronjob.go [hunk 4]
```diff
@@ -51,59 +78,60 @@ func NewCronJob(client mgrclient.Client, namespace string, filter filter.Filter)
...
 	cronJobList.Items = []batchv1beta1.CronJob{cronJob}
 	resultList := c.addExtendInfo(cronJobList)
 	if len(resultList) == 0 {
 		clog.Error("can not parse cronjob %s/%s", c.namespace, name)
-		return nil
+		return nil, err
 	}
 
-	return resultList[0].(filter.K8sJson)
+	return resultList[0].(filter.K8sJson), err
```
Comment (weilaaa) [inline pkg/apiserver/cubeapi/resourcemanage/resources/cronjob/cronjob.go:134]: add // fixme

File: pkg/utils/filter/filter_test.go [hunk 5]
```diff
@@ -288,10 +288,12 @@ func TestGetDeepValue(t *testing.T) {
 
 	items := result["items"].([]interface{})
 
-	value := GetDeepValue(items[0], "metadata.labels.hello")
-	assert.Equal("world", value)
-	value = GetDeepValue(items[0], "metadata.labels.hello1")
-	assert.Equal("", value)
+	value, err := GetDeepValue(items[0], "metadata.labels.hello")
```
Comment (weilaaa) [inline pkg/utils/filter/filter_test.go:291]: add case to test arry

**Review by JiahuiZhao11 — COMMENTED** [review]
LGTM

**Review by weilaaa — APPROVED** [review]
LGTM

**Pull request merged**